app/uk/gov/hmrc/agentsubscriptionfrontend/views/task_list.scala.html

```diff
@@ -17,81 +17,45 @@
     <h1 class="heading-xlarge margin-bottom-40">@Messages("task-list.header")</h1>
 
     <p>@Messages("task-list.subheader")</p>
 
     <ol class="list">
-        <li>
-            <h2>
-                <span class="app-task-list__section-number">@Messages("task-list.1.number")</span>@Messages("task-list.1.header")
-            </h2>
-            <ul class="app-task-list__items">
-                <li class="app-task-list__item">
-                    @if(taskListFlags.cannotEditAmls) {
-                        @Messages("task-list.1.amls")
-                    } else {
-                        <a href=@routes.AMLSController.showAmlsRegisteredPage()>@Messages("task-list.1.amls")</a>
-                    }
-                    @if(taskListFlags.amlsTaskComplete) { @completedMark }
-                </li>
-            </ul>
-        </li>
-        <li>
-            <h2>
-                <span class="app-task-list__section-number">@Messages("task-list.2.number")</span>@Messages("task-list.2.header")
-            </h2>
-            <ul class="app-task-list__items">
-                <li class="app-task-list__item">
-                    @if(taskListFlags.mustCreateCleanCreds) {
-                        <a href=@routes.BusinessIdentificationController.showCreateNewAccount()>@Messages("task-list.2.create")</a>
-                    } else {
-                        @Messages("task-list.2.create")
-                    }
-                    @if(taskListFlags.createTaskComplete) { @completedMark }
-                </li>
-            </ul>
-        </li>
-        <li>
-            <h2>
-                <span class="app-task-list__section-number">@Messages("task-list.3.number")</span>@Messages("task-list.3.header")
-            </h2>
-            <ul class="app-task-list__items">
-                <li class="app-task-list__item">
-                    @if(taskListFlags.mustCheckAnswers) {
-                        <a href=@routes.SubscriptionController.showCheckAnswers()>@Messages("task-list.3.header")</a>
-                    } else {
-                        @Messages("task-list.3.header")
-                    }
-                    @if(taskListFlags.checkAnswersComplete) { @completedMark }
-                </li>
-            </ul>
-        </li>
-        <li>
-            <h2>
-                <span class="app-task-list__section-number">@Messages("task-list.4.number")</span>@Messages("task-list.4.header")
-            </h2>
-            <ul class="app-task-list__items">
-                <li class="app-task-list__item">
-                @if(taskListFlags.checkAnswersComplete) {
-                    <a href="@routes.SubscriptionController.beginMapping()">@Messages("task-list.4.copy")</a>
-                } else {
-                    @Messages("task-list.4.copy")
-                }
-                </li>
-            </ul>
-        </li>
+    @for(task <- tasks) {
+      @listItem(tasks.indexOf(task) + 1, task.taskKey, task.showLink, task.isComplete, task.link)
```

Comment:
Could you use zipWithIndex here instead of the indexOf() thing?

Reply:
can't really use zipWithIndex because it wants to start at 1 not 0 but can use zip(Stream from 1)